Sources/StreamChatUI/CommonViews/InputTextView/InputTextView.swift

```diff
@@ -2,12 +2,28 @@
 // Copyright © 2021 Stream.io Inc. All rights reserved.
 //
 
+import MobileCoreServices
 import StreamChat
 import UIKit
 
+/// The delegate of the `InputTextView` that notifies when an `UIImage` is pasted.
+public protocol InputTextViewImagePasteDelegate: AnyObject {
```

Comment:
```suggestion
public protocol InputTextViewDelegate: AnyObject {
```

Reply:
Why would we want this? The `InputTextView` is already a subclass of `UITextView` so we already have the `delegate: UITextViewDelegate?` out of the box.

Reply:
I think we should keep the delegates separate as @nuno-vieira suggests. But yes, we can make the other delegate more generic, like: `InputTextViewClipboardAttachmentDelegate`. This will currently contain only one method for the image, and we may add more in the future as needed. Will this be fine?

Reply:
With a protocol inherited from the base one there's no need in additional delegate field (the casted delegate is computed and can be private) and no need to assign multiple delegates to handle all available callbacks. This is how delegates are extended in UIKit (UITableViewDelegate: ScrollViewDelegate) and I find it useful, tbh.

If customer decide to have a custom `InputTextView` with additional callback it'd would be simple:

```swift
protocol MyInputTextViewDelegate: InputTextViewDelegate {
 func callback()
}

class MyInputTextView: InputTextView {
 func smthDidHappen() {
 (delegate as? MyInputTextViewDelegate).callback()
 }
}

class MyComposerVC: ComposerVC, MyInputTextViewDelegate {
  func callback() { ... }
}
```

Reply:
it works great in message list with custom attachments but here custom delegate is an edge case. both approaches work for me so feel free to go with one you like more 👍